cristin-commons/src/main/java/no/unit/nva/cristin/model/QueryParameterKey.java

```diff
@@ -0,0 +1,183 @@
+import static no.unit.nva.cristin.common.ErrorMessages.INVALID_URI_MESSAGE;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_PER_PAGE_PARAM;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_QUERY_NAME_PARAM;
+import static no.unit.nva.cristin.model.Constants.PARENT_UNIT_ID;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_DATE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_LANGUAGE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_NON_EMPTY;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_NUMBER;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_TITLE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_URL;
+import static no.unit.nva.cristin.model.Constants.PROJECTS_PATH;
+import static no.unit.nva.cristin.model.Constants.PROJECT_PATH_NVA;
+import static no.unit.nva.cristin.model.Constants.QUERY_PARAMETER_LANGUAGE;
+import static no.unit.nva.cristin.model.QueryParameterKey.QueryParameterConstant.ERROR_MESSAGE_INVALID_CHARACTERS;
+import static no.unit.nva.model.Organization.ORGANIZATION_IDENTIFIER_PATTERN;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public enum QueryParameterKey {
+    INVALID(null),
+    PATH_IDENTITY(JsonPropertyNames.IDENTIFIER, null, PATTERN_IS_NON_EMPTY),
+    PATH_ORGANISATION(PARENT_UNIT_ID,
+        JsonPropertyNames.ORGANIZATION,
+        ORGANIZATION_IDENTIFIER_PATTERN,
+        ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID_FOUR_NUMBERS,
+        true),
+    PATH_PROJECT(PROJECTS_PATH, PROJECT_PATH_NVA, PATTERN_IS_NUMBER),
+    APPROVAL_REFERENCE_ID(JsonPropertyNames.PROJECT_APPROVAL_REFERENCE_ID),
+    APPROVED_BY(JsonPropertyNames.PROJECT_APPROVED_BY),
+    BIOBANK(JsonPropertyNames.BIOBANK_ID),
+    FUNDING(JsonPropertyNames.FUNDING),
+    FUNDING_SOURCE(JsonPropertyNames.FUNDING_SOURCE),
+    GRANT_ID("project_code", null, PATTERN_IS_NUMBER),
+    INSTITUTION(JsonPropertyNames.INSTITUTION),
+    KEYWORD(JsonPropertyNames.PROJECT_KEYWORD),
+    LANGUAGE(QUERY_PARAMETER_LANGUAGE, JsonPropertyNames.LANGUAGE, PATTERN_IS_LANGUAGE),
+    LEVELS(JsonPropertyNames.LEVELS, JsonPropertyNames.DEPTH, PATTERN_IS_NUMBER),
+    MODIFIED_SINCE(JsonPropertyNames.PROJECT_MODIFIED_SINCE, null, PATTERN_IS_DATE),
+    NAME(JsonPropertyNames.TITLE,
+         CRISTIN_QUERY_NAME_PARAM,
+         PATTERN_IS_TITLE,
+         String.format(ERROR_MESSAGE_INVALID_CHARACTERS,JsonPropertyNames.TITLE),
+         true),
+    ORGANIZATION(PARENT_UNIT_ID,JsonPropertyNames.ORGANIZATION,PATTERN_IS_URL,INVALID_URI_MESSAGE,true),
+    PARTICIPANT(JsonPropertyNames.PROJECT_PARTICIPANT),
+    PROJECT_MANAGER(JsonPropertyNames.PROJECT_MANAGER),
+    PROJECT_UNIT(JsonPropertyNames.UNIT),
+    QUERY(JsonPropertyNames.QUERY,
```

Comment:
query is not a valid cristin param

Reply:
Query blir oversatt til titlle eller grantid alt etter hva query inneholder (som den gamle koden)